src/core/c-frame.c

```diff
@@ -469,11 +486,11 @@ void Rebind_Frame(REBSER *src_frame, REBSER *dst_frame)
 
 //
 //  Make_Object: C
-// 
-// Create an object from a parent object and a spec block.
+//
+// Create an object from a parent object, a spec block and body block.
 // The words within the resultant object are not bound.
 //
-REBSER *Make_Object(REBSER *parent, REBVAL value[])
+REBSER *Make_Object(REBSER *parent, REBSER *spec, REBVAL value[])
```

Comment:
What we're going to see for Make_Object will likely very much parallel what we see for Make_Function today.  It will probably be a shared generator that implements both the MAKE OBJECT! (which will have no awareness of things like SELF, for instance) *and* the native-optimized version of the OBJECT generator, which I imagine will have other features included.

So if you haven't kind of absorbed how Make_Function works, you might want to look at the comments there:

https://github.com/metaeducation/ren-c/blob/990816d46a4fad14e79cdd2fbc8dddd3f1b64831/src/core/c-function.c#L381

There are some ideas in there that I think I would like to see repeated.  For instance to make it so there is *always* a spec, but reuse some common global read-only blocks if a pattern is very common.  We will probably see `[]` and `[self:]` for instance, much like how those are common in the function builders.

If there is always a spec then that is easier than handling a NULL case.